src/async-stream.js

```diff
@@ -0,0 +1,124 @@
+(function (global) {
+  'use strict';
+
+  const Iteration = (value, done) => {
+    return {value, done};
+  }
+
+  global.Iteration = Iteration;
+
+  const Stream = (syn, ack) => {
+    return {
+      __proto__: Stream.prototype,
+      next(value) {
+        ack.put(Iteration(value, false))
+        return syn.get();
+      },
+      return(value) {
+        ack.put(Iteration(value, true))
+        return syn.get();
+      },
+      throw(error) {
+        ack.put(Promise.reject(error))
+        return syn.get();
+      },
+    }
+  }
+
+  Stream.prototype = {
```

Comment:
This is a good example of why it is important to initialize with defineProperty rather than assignment. This assignment overrides Function.prototype. If Function.prototype is a non-writable data property, then this override-by-assignment will fail with a thrown error.